overops-functions/src/main/java/com/takipi/udf/microsoftteams/MicrosoftTeamsChannelFunction.java

```diff
@@ -0,0 +1,195 @@
+package com.takipi.udf.microsoftteams;
+
+import com.google.common.base.Strings;
+import com.google.gson.Gson;
+import com.takipi.api.client.ApiClient;
+import com.takipi.api.client.data.metrics.Graph;
+import com.takipi.api.client.data.service.SummarizedService;
+import com.takipi.api.client.data.view.SummarizedView;
+import com.takipi.api.client.request.event.EventRequest;
+import com.takipi.api.client.request.event.EventsRequest;
+import com.takipi.api.client.result.event.EventResult;
+import com.takipi.api.client.result.event.EventsResult;
+import com.takipi.api.client.util.event.EventUtil;
+import com.takipi.api.client.util.validation.ValidationUtil;
+import com.takipi.api.client.util.view.ViewUtil;
+import com.takipi.api.core.url.UrlClient;
+import com.takipi.common.util.CollectionUtil;
+import com.takipi.udf.ContextArgs;
+import org.joda.time.format.DateTimeFormatter;
+import org.joda.time.format.ISODateTimeFormat;
+
+import java.util.Base64;
+import java.util.List;
+
+import static com.takipi.udf.microsoftteams.MicrosoftTeamsUtil.*;
+
+public class MicrosoftTeamsChannelFunction {
+
+    public static final int MINUTES_TIME_SPAN = 5;
+
+    public static String validateInput(String rawInput) {
+        return getInput(rawInput).toString();
+    }
+
+    private static MicrosoftTeamsInput getInput(String rawInput) {
+        if (Strings.isNullOrEmpty(rawInput))
+            throw new IllegalArgumentException("Input is empty");
+
+        MicrosoftTeamsInput input;
+
+        try {
+            input = MicrosoftTeamsInput.of(rawInput);
+        } catch (Exception e) {
+            throw new IllegalArgumentException(e.getMessage(), e);
+        }
+
+        input.checkUrl();
+
+        return input;
+    }
+
+    public static void execute(String rawContextArgs, String rawInput) {
+        try {
+            executeImplementation(rawContextArgs, rawInput);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    private static void executeImplementation(String rawContextArgs, String rawInput) {
+        logUDFInput(rawContextArgs, rawInput);
+        MicrosoftTeamsInput input = getInput(rawInput);
+        ContextArgs args = (new Gson()).fromJson(rawContextArgs, ContextArgs.class);
+
+        if (!args.validate())
```

Comment:
Please have `{}` surrounding if code blocks, even if it's just a single line,.
In this case:

```
if (!args.validate()) {
    throw new IllegalArgumentException("Bad context args: " + rawContextArgs);
}
if (!args.eventValidate()) {
    return;
}
```